### PR TITLE
Remove E2E and Gitlab related code

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

Remove all code concerning E2E testing, and referencing Gitlab. Also cleanup CircleCI references.

After this is merge, mattermod won't react anymore to `/e2e-*` slash command comments, or labels.

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-7229 (for the monorepo)
https://mattermost.atlassian.net/browse/CLD-6961 (for mobile)

Desktop E2E testing was already disabled, but not cleaned up

### CODE REVIEW

**Review by phoinixgrr — APPROVED** [review]

File: server/pull_request_handler.go [hunk 1]
```diff
@@ -96,11 +96,6 @@ func (s *Server) pullRequestEventHandler(w http.ResponseWriter, r *http.Request)
 			return
 		}
 
-		if (pr.RepoName == s.Config.E2EServerReponame || pr.RepoName == s.Config.E2EWebappReponame) && contains(s.Config.E2ETriggerLabel, *event.Label.Name) {
-			mlog.Info("Label to run e2e tests", mlog.Int("pr", event.PRNumber), mlog.String("repo", pr.RepoName), mlog.String("label", *event.Label.Name))
-			go s.triggerE2ETestFromLabel(pr)
-		}
```
Comment (hanzei) [inline server/pull_request_handler.go:102]: Can we also remove the label from https://github.com/mattermost/mattermost ?

Comment (mvitale1989) [reply server/pull_request_handler.go:102]: Absolutely :+1: I can do that already

Comment (mvitale1989) [reply server/pull_request_handler.go:102]: Ah wait, those are not only for triggering the test, but also for marking PRs for QA testers. The two labels that exist right now should remain as they are:
- `3: QA Review`
- `QA Deferred`

Comment (mvitale1989) [reply server/pull_request_handler.go:102]: Also for best compatibility i'd port this label-triggered functionality to GHA as well, i'll go ahead and prepare a PR for that

**Review by saturninoabril — APPROVED** [review]
Nice, thanks @mvitale1989!

**Comment by toninis** [comment]
🚀 Removing code is always good 😄

**Pull request merged**